packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -633,6 +668,43 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
     };
   }
 
+  private createArtifactConfig(props: CanaryProps): CfnCanary.ArtifactConfigProperty | undefined {
+    if (!props.artifactS3EncryptionMode) {
+      return undefined;
+    }
```

Comment:
What would be the behaviour if the user provides `props.artifactS3KmsKey` but not `props.artifactS3EncryptionMode`?

Reply:
Thanks.
I tried this case and Canary uses SSE-S3 (CMK is ignored).

But, I think this case should throw an error, so I've added a validation.

Reply:
I don't think throwing an error is the best route here, since if the user provides a KMS key, it's likely they'll want to use SSE-KMS, and it would be more intuitive for us to set it for them (see: [Avoid Errors if Possible](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md#avoid-errors-if-possible)).

Reply:
Thanks.
In that case, I have modified it to automatically use SSE KMS.​​​​​​​​​​​​​​​​